packages/portfolio-contract/src/portfolio.flows.ts

```diff
@@ -588,8 +697,14 @@ export const rebalance = async (
     offerArgs,
   );
 
-  if (keys(proposal.want).length > 0) {
-    throw Error('TODO: withdraw');
+  if (!('give' in proposal) || Object.keys(proposal.give).length === 0) {
+    // throw Error('TODO: withdraw');
+    if (!('want' in proposal)) {
+      throw Error('rebalance proposal must have give or want');
+    }
+    const pos = kit.manager.provideUSDNPosition(); // TODO: get num from offerArgs?
```

Comment:
We're now trying to reserve TODO to mean "won't work at all in production" vs XXX for "could / should be nicer".

```suggestion
    const pos = kit.manager.provideUSDNPosition(); // XXX: get num from offerArgs?
```